raspberry/SmartHome/arduinodevices/src/test/java/smarthome/raspberry/arduinodevices/server/WebServerTest.kt

```diff
@@ -0,0 +1,79 @@
+package smarthome.raspberry.arduinodevices.server
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import fi.iki.elonen.NanoHTTPD
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+import smarthome.library.common.BaseController
+import smarthome.library.common.DeviceChannelOutput
+import smarthome.raspberry.arduinodevices.ArduinoDevice
+import smarthome.raspberry.arduinodevices.StringValueState
+
+class WebServerTest {
+
+    private val session: NanoHTTPD.IHTTPSession = mock()
+    private val deviceChannelOutput: DeviceChannelOutput = mock()
+    private val handler = WebHandler(deviceChannelOutput)
+    private val controller = BaseController("")
+
+    @Before
+    fun setUp() {
+        runBlocking {
+            whenever(deviceChannelOutput.findController(any())).thenReturn(controller)
```

Comment:
DeviceChannelOutput does not have findController function

Reply:
it does